src/lib/features/listening-history/model/store.ts

```diff
@@ -12,16 +26,36 @@ export const listeningHistory = persistentWritable<number[]>([], {
  * Function that adds a podcast id to the beginning of listening history.
  * It removes any previous occurrences of the same podcast.
  *
- * @param podcastId {string} – an URl to add to history
+ * @param podcastId {string} – an id to add to history
  */
-export const addToListeningHistory = (podcastId: number) => {
+export const addToLocalListeningHistory = (podcastId: number) => {
   listeningHistory.update((history) => {
     return [podcastId, ...history.filter((id) => id !== podcastId)].slice(0, 6);
   });
 };
 
-currentlyPlayingEpisode.subscribe(($episode) => {
+/**
+ * Function that add a row into history SB table.
+ *
+ * @param podcastId {string} – an id to add to history
+ */
+export const addToCloudListeningHistory = async (podcastId: number) => {
+  const u = get(user);
+  if (!u) return;
+  await supabaseClient.from('history').insert([
+    {
+      // eslint-disable-next-line camelcase
+      podcast_id: podcastId,
+      // eslint-disable-next-line camelcase
+      user_id: u.id,
```

Comment:
suggestion (non-blocking): looks like there's a way to let Supabase figure out the user ID itself: https://github.com/supabase/supabase/discussions/4368#discussioncomment-1769320